src/Components/itemDetail.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import '../App.css';
+import { useParams } from 'react-router-dom';
+import { getLocalToken } from '../lib/token.js';
+import { FirestoreDocument } from 'react-firestore';
+import { Link } from 'react-router-dom';
+
+function ItemDetailPage() {
+  let { docId } = useParams();
+  // date formater
+
+  // calculate days until next purchase
+  const getFormattedDate = date => {
+    const seconds = date.seconds;
+    const year = Math.floor(seconds / 31536000);
+    const nummonths = Math.floor((seconds % 31536000) / 2628000);
+    const numdays = Math.floor(((seconds % 31536000) % 2628000) / 86400);
+
+    const month = nummonths.toString().padStart(2, '0');
+    const day = numdays.toString().padStart(2, '0');
+
+    return month + '/' + day + '/' + year;
```

Comment:
This is totally fine, but remember that you can also do:

```js
return `${month}/${day}/${year}`
```

And get the same result, this is a pattern that you'll start to see in other _newish_ codebases :)